source/common/http/hash_policy.cc

```diff
@@ -33,21 +36,33 @@ class HeaderHashMethod : public HashMethodImplBase {
     }
   }
 
+  const std::string HEADER_VALUE_SEPARATOR = ",";
+
   absl::optional<uint64_t> evaluate(const Network::Address::Instance*,
                                     const RequestHeaderMap& headers,
                                     const HashPolicy::AddCookieCallback,
                                     const StreamInfo::FilterStateSharedPtr) const override {
     absl::optional<uint64_t> hash;
 
-    // TODO(mattklein123): Potentially hash on all headers.
     const auto header = headers.get(header_name_);
     if (!header.empty()) {
-      if (regex_rewrite_ != nullptr) {
-        hash = HashUtil::xxHash64(regex_rewrite_->replaceAll(header[0]->value().getStringView(),
-                                                             regex_rewrite_substitution_));
-      } else {
-        hash = HashUtil::xxHash64(header[0]->value().getStringView());
+      std::vector<std::string> header_values;
```

Comment:
WDYT of optimizing for the fast case where there's no regex rewriter, and build a `vector<string_view>`. Then you can make a special case for when there is a rewriter and you need to make std::string copies of all the headers.

Reply:
Yea, I can do that, but I need another version xxHash64 for accept vector<string_view>, is it ok for you?

Reply:
Yes that makes sense. I guess if one if we're doing this regex transform it will be for all strings.

I wouldn't make this part of the xxHash64 layer -- just call it directly.

Reply:
got it, thanks

Reply:
ah...sorry, I forget I still need the separator at line 55. so I have to make copies of all the headers with separator. or I didn't get what you mean in line 52.

Reply:
why do you need any separators at all?

note that hash(["ab"]) != hash(["a", "b"])

Reply:
> note that hash(["ab"]) != hash(["a", "b"])

ah, i got it, I missunderstand this.